index.js

```diff
@@ -7,6 +7,43 @@ const PLAYING = 2;
 const MODE_LERP = 0;
 const MODE_HERMITE = 1;
 
+const vectorPool = [];
+const quatPool = [];
+const framePool = [];
+
+const getPooledVector = () => vectorPool.shift() || new THREE.Vector3();
+const getPooledQuaternion = () => quatPool.shift() || new THREE.Quaternion();
+
+const getPooledFrame = () => {
+  let frame = framePool.shift() || { position: null, velocity: null, scale: null, quaternion: null, time: 0 };
```

Comment:
JIT will be happier if frames are always the same shape - i.e. don't start with `position: null` and don't write `f.position = null` when you free the frame. 
Since you are only using `getPooledVector` and `getPooledQuaternion` when you `getPooledFrame`, it is an easy adjustment to only have a pool of frames where each frame holds onto its vectors and quaternion. I think this is preferable.

Reply:
this is a lot smarter, thanks! i wrote the vec/quat pooling first and added the frames. the `= null` bit was to unroot the objects so they'd GC but having the whole frame be the unit makes way more sense.